From cb021ad04cb68acc18ad0f578ebe3bfaf99b904a Mon Sep 17 00:00:00 2001
From: Josh Fazekas <Josh.Fazekas@Intel.com>
Date: Fri, 13 Jun 2025 21:11:43 +0300
Subject: [PATCH] [SW-229355] g3: pldm: SRAM fatal repair errors

Set sensor ID 339 - composite sensor index 2 - SRAM fatal repair errors
to return a default value of Operational Fault Status Normal

Change-Id: I00e9776124f54ccb53651e4f1acd748d13a62af3
Signed-off-by: Josh Fazekas <Josh.Fazekas@Intel.com>
---

diff --git a/arch/gaudi3/common/ras/pldm/sensord_handlers.c b/arch/gaudi3/common/ras/pldm/sensord_handlers.c
index 1a51052..5a0080c 100644
--- a/arch/gaudi3/common/ras/pldm/sensord_handlers.c
+++ b/arch/gaudi3/common/ras/pldm/sensord_handlers.c
@@ -923,8 +923,9 @@
 		count = 0;
 	sample->composite[COMP_SENSOR_FATAL_RMA_ECC_SRAM_FATAL] = check_rma_error_state(count);
 
-	/* TODO: pull SRAM repair data once filled in */
-	sample->composite[COMP_SENSOR_FATAL_RMA_ECC_SRAM_FATAL_REPAIR] = 0;
+	/* TODO: report normal for now - pull SRAM repair data once filled in */
+	sample->composite[COMP_SENSOR_FATAL_RMA_ECC_SRAM_FATAL_REPAIR] =
+		PLDM_STATE_SET_OPERATIONAL_FAULT_STATUS_NORMAL;
 
 	ret = hl_mem_info_rd_field(&count, hbm_fatal_events);
 	if (ret)
